oops/oops.go

```diff
@@ -149,6 +172,19 @@ func Frames(err error) [][]Frame {
 			}
 
 			file := frame.File
+
+			for prefixToSkip := range filePrefixesToSkip {
+				if strings.HasPrefix(file, prefixToSkip) {
+					framesSkipped = true
+					break
```

Comment:
There are already kind of a lot of helper functions involved here, but it might be worthwhile to extract the stack-parsing for loop into a separate function. That way, you could just return early here instead of needing the conditional `break` in the outer scope below.

Reply:
I tried, but there's a lot of state external to the for loop that's being accessed which makes this messy to refactor. I think keeping this as is makes it more readable and cleaner in its current state.

Reply:
What about something like the following? It's really just the double break below that feels worthwhile to avoid.

```
filePrefixesToSkipMap := filePrefixesToShortCircuit.Load().(map[string]struct{})
file := frame.File

if mapContainsKeyWithPrefix(filePrefixesToSkipMap, file) {
  // Skip appending this and all other frames from this stack.
  break
}
```